pageserver/src/layered_repository.rs

```diff
@@ -35,10 +35,10 @@ use self::metadata::{metadata_path, TimelineMetadata, METADATA_FILE_NAME};
 use crate::config::PageServerConf;
 use crate::page_cache;
 use crate::relish::*;
-use crate::remote_storage::{schedule_timeline_checkpoint_upload, schedule_timeline_download};
+use crate::remote_storage::schedule_timeline_checkpoint_upload;
```

Comment:
I've not noticed anything changing to the upload checks: does that mean, that we now can schedule whatever arbitrary upload for the same timeline, even if it's already downloaded/not loaded, etc?

One of the main benefits of commonly accessible timeline sync state should be that IMO, so could be reused.

Reply:
It is possible to implement this check on top of what is provided by this PR. It is enough to check the flag inside remote timeline state. It is missing intentionally, and can be done as part of remaining TODO's or separately

Reply:
added to the follow up list in #977